Adding projects for RecoveryServices.SiteRecovery service.

avneeshrai opened the pull request:
It's corresponding to new swagger spec which is under PR https://github.com/Azure/azure-rest-api-specs/pull/1209
Intent is to get early feedback here.

<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!--
Please add an informative description that covers that changes made by the pull request.

If you are regenerating your SDK based off of a new swagger spec, please add the link to the corresponding swagger spec pull request that has been merged in the azure-rest-api-specs repository
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

avneeshraiusit committed:
Adding projects for RecoveryServices.SiteRecovery service.

msftclas commented:
@avneeshrai,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

avneeshraiusit committed:
Changes corresponding to renaming GetTrackAsyncOperation to GetOperationResults in operation id

avneeshraiusit committed:
Changes corresponding to update in swagger spec PR:
1) Removing operationresults apis.
2) Introducing operations api.
3) Fixing issue in a patch model.
4) Changing few fields to string from boolean.

avneeshrai commented:
Can someone review this PR please. Corresponding swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/1209 is merged.

I have removed do not merge tag.

shahabhijeet commented on src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs:
```diff
@@ -0,0 +1,47 @@
+// Code generated by Microsoft (R) AutoRest Code Generator 0.16.0.0
+// Changes may cause incorrect behavior and will be lost if the code is
```
@avneeshrai missing license header.
Chances are your swagger does not have the right License Header Constant, please double check, I believe the constant that you need to use in your swagger is "MIT-no-version"

avneeshrai replied:
@shahabhijeet can you please point to some example spec on what exacty needs to be done here
I tried searching MIT-no-version in azure-rest-api-spec repository but couldnt find any instance of it

avneeshrai replied:
Fixed in the latest commit

shahabhijeet commented on src/SDKs/RecoveryServices.SiteRecovery/Management.RecoveryServices.SiteRecovery/Generated/IOperationsOperations.cs:
```diff
@@ -0,0 +1,47 @@
+// Code generated by Microsoft (R) AutoRest Code Generator 0.16.0.0
```
@avneeshrai please use latest AutoRest (at least 1.0)

avneeshrai replied:
even after doing autorest --latest or autorest --reset 
the version that i get is 0.16.0.0.
How to i get 1.0?

avneeshraiusit committed:
Generated files with proper license headers.

avneeshrai commented:
@shahabhijeet Please review and merge the change in swagger spec as you had suggested in the PR: https://github.com/Azure/azure-rest-api-specs/pull/1227

avneeshraiusit committed:
Changes corresponding to ixing case of discriminator field for FabricSpecificUpdateNetworkMappingInput

avneeshraiusit committed:
Generating files with latest autorest version.

avneeshraiusit committed:
Adding new tests.

avneeshraiusit committed:
Removing redundant dependencies.

avneeshrai commented on src/SDKs/RecoveryServices.SiteRecovery/RecoveryServices.SiteRecovery.Tests/RecoveryServices.SiteRecovery.Tests.csproj:
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
@shahabhijeet as discussed offline...we dont have dependency on Resource manager in our projects. Have removed all the stale references that we had.
